### PR TITLE
chore: measure response times of Companion endpoints in blackbox tests

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- measure response times of Companion endpoints in blackbox tests
- Example:

image

**Related issue(s)**

- https://github.com/kyma-project/kyma-companion/issues/29

### CODE REVIEW

File: tests/blackbox/src/common/metrics.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class Metrics:
+    """Singleton class to store metrics."""
+
+    __instance = None
```
Comment (friedrichwilken) [inline tests/blackbox/src/common/metrics.py:4]: Q: Why does this have a double-underscore?

File: tests/blackbox/src/common/metrics.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        if Metrics.__instance is None:
+            Metrics()
+        return Metrics.__instance
+
+    def __init__(self):
+        """Virtually private constructor."""
+        if Metrics.__instance is not None:
+            raise Exception("This class is a singleton!")
+        else:
+            Metrics.__instance = self
```
Comment (friedrichwilken) [inline tests/blackbox/src/common/metrics.py:20]: Q: Can you explain this design?

File: tests/blackbox/src/common/metrics.py (hunk 1 above)
Comment (mfaizanse) [reply tests/blackbox/src/common/metrics.py:4]: https://dbader.org/blog/meaning-of-underscores-in-python

File: tests/blackbox/src/common/metrics.py (hunk 2 above)
Comment (mfaizanse) [reply tests/blackbox/src/common/metrics.py:20]: https://refactoring.guru/design-patterns/singleton

**Pull request merged**